src/pages/search/Search.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import { graphql, StaticQuery } from 'gatsby';
+import { Link } from 'gatsby';
+
+import queryString from 'query-string';
+import { Index } from 'elasticlunr';
+
+import Layout from '../../components/Layout';
+import Flex from '../../components/Flex';
+import { colors, media } from '../../components/theme';
+
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchToggle: true,
+      searchValue: '',
+      results: []
+    };
+    this.toggleSearch = toggle => this._toggleSearch(toggle);
+    this.search = (query, idx) => this._search(query, idx);
+    this.getOrCreateIndex = () => this._getOrCreateIndex();
+  }
+
+  componentDidMount = () => {
+    const values = queryString.parse(this.props.location.search);
+    const idx = Index.load(this.props.data.search.index);
+
+    if (values.q) {
+      this.setState({ searchValue: values.q }, () => {
+        this.search(this.state.searchValue, idx);
+      });
+    }
+  };
+
+  componentDidUpdate = prevprops => {
+    const values = queryString.parse(this.props.location.search);
+    const idx = Index.load(this.props.data.search.index);
+
+    if (values.q) {
+      if (this.state.searchValue !== values.q) {
+        this.setState({ searchValue: values.q }, () => {
```

Comment:
There is some duplication in componentDidMount and componentDidUpdate lifecycle events. Maybe extract in a function?